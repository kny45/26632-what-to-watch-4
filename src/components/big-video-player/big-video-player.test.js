import React from "react";
import renderer from "react-test-renderer";
import BigVideoPlayer from "./big-video-player.jsx";

describe(`Render BigVideoPlayer`, () => {
  it(`Should BigVideoPlayer render correctly`, () => {
    const movieCard = {
      name: `Fantastic Beasts: The Crimes of Grindelwald`,
      previewImage: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    };
    const children = <div className="children-mock-component" />;

    const tree = renderer
      .create(
          <BigVideoPlayer
            movieCard = {movieCard}
            isPlaying = {false}
            onPlayButtonClick = {() => {}}
            onFullScreenButtonClick = {() => {}}
            getPlaybackProgress = {() => {}}
            getTimeLeft = {() => {}}
            onExitButtonClick = {() => {}}
          >
            {children}
          </BigVideoPlayer>, {
            createNodeMock: () => {
              return {};
            }
          })
        .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
