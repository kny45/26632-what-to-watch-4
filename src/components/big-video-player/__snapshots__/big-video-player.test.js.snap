// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render BigVideoPlayer Should BigVideoPlayer render correctly 1`] = `
<div
  className="player"
>
  <div
    className="children-mock-component"
  />
  <a
    className="player__exit"
    href="/films/1"
    onClick={[Function]}
    type="button"
  >
    Exit
  </a>
  <div
    className="player__controls"
  >
    <div
      className="player__controls-row"
    >
      <div
        className="player__time"
      >
        <progress
          className="player__progress"
          max="100"
        />
        <div
          className="player__toggler"
          style={
            Object {
              "left": "undefined%",
            }
          }
        >
          Toggler
        </div>
      </div>
      <div
        className="player__time-value"
      />
    </div>
    <div
      className="player__controls-row"
    >
      <button
        className="player__play"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="19"
          viewBox="0 0 19 19"
          width="19"
        >
          <use
            xlinkHref="#play-s"
          />
        </svg>
        <span>
          Play
        </span>
      </button>
      <div
        className="player__name"
      >
        Fantastic Beasts: The Crimes of Grindelwald
      </div>
      <button
        className="player__full-screen"
        onClick={[Function]}
        type="button"
      >
        <svg
          height="27"
          viewBox="0 0 27 27"
          width="27"
        >
          <use
            xlinkHref="#full-screen"
          />
        </svg>
        <span>
          Full screen
        </span>
      </button>
    </div>
  </div>
</div>
`;
